### PR TITLE
a command line utility for silencing

### PR DESCRIPTION
**Pull request by kylebrandt**

### CODE REVIEW

File: cmd/silence/main.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	}
+	b, err := json.Marshal(s)
+	if err != nil {
+		log.Fatal(err, s)
+	}
+	resp, err := http.Post("http://"+*flagHost+"/api/silence/set", "image/jpeg", bytes.NewBuffer(b))
+	if err != nil {
+		log.Fatal(err, s)
+	}
+	c, err := ioutil.ReadAll(resp.Body)
```
Comment (captncraig) [inline cmd/silence/main.go]: defer resp.Body.Close()

Not really required in such a short app, but a good habit nonetheless.

File: cmd/silence/main.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		log.Fatal(err)
+	}
+	if resp.StatusCode != 200 {
+		log.Fatalf("%s", c)
+	}
+	if s.Alert == "" {
+		s.Alert = "None"
+	}
+	if s.Tags == "" {
+		s.Alert = "None"
```
Comment (captncraig) [inline cmd/silence/main.go]: I think you meant s.Tags.

File: cmd/silence/main.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	now := time.Now().UTC()
+	d, err := time.ParseDuration(*flagDuration)
+	if err != nil {
+		log.Fatal(err)
+	}
+	end := now.Add(d)
+	if *flagForget != "" {
+		*flagForget = "true"
+	}
+	s := struct {
```
Comment (captncraig) [inline cmd/silence/main.go]: Is this struct not defined somewhere else in bosun?

File: cmd/silence/main.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		Start   string `json:"start"`
+		End     string `json:"end"`
+		Tags    string `json:"tags"`
+		Alert   string `json:"alert"`
+		Message string `json:"message"`
+		Confirm string `json:"confirm"`
+		Forget  string `json:"forget"`
+	}{
+		un,
+		now.Format("2006-01-02 15:04:05 MST"),
```
Comment (captncraig) [inline cmd/silence/main.go]: There should be a constant for this format string somewhere.

File: cmd/silence/main.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+		*flagAlert,
+		*flagMessage,
+		"confirm",
+		*flagForget,
+	}
+	b, err := json.Marshal(s)
+	if err != nil {
+		log.Fatal(err, s)
+	}
+	resp, err := http.Post("http://"+*flagHost+"/api/silence/set", "image/jpeg", bytes.NewBuffer(b))
```
Comment (captncraig) [inline cmd/silence/main.go]: image/jpeg? wtf?

File: cmd/silence/main.go (hunk 4 above)
Comment (kylebrandt) [inline cmd/silence/main.go]: There isn't

File: cmd/silence/main.go (hunk 3 above)
Comment (kylebrandt) [inline cmd/silence/main.go]: It is not, the API to create silences differs a fair amount from the silence struct itself, I started to change this, but then with all the date parsing etc I decided it wasn't worth the refactor

**Comment by captncraig** [comment]
LGTM

**Pull request merged**